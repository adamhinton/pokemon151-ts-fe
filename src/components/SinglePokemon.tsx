import React, { ReactElement } from "react";
import { Pokemon } from "../data";
import styled from "styled-components";

interface Props {
  poke: Pokemon;
  // isShiny will be added later
  isShiny: Boolean;
}

const PokemonCard = styled.article`
  border: 1px solid black;
  box-sizing: border-box;
  width: 12rem;
  margin: 2%;
  background-color: #bef8f8;
`;

const PokeID = styled.span`
  font-size: 70%;
`;

const CardItem = styled.div`
  margin: 3%;
`;

const PokeImageContainer = styled(CardItem)`
  height: 171px;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const PokeImage = styled.img`
  width: 90%;
`;

const SinglePokemon = (props: Props): ReactElement => {
  const { poke, isShiny } = props;
  const { name, type, dexEntry, height, weight, id, num } = poke;

  return (
    <PokemonCard data-testid="pokemon-card">
      <CardItem as="h2" className="poke-name card-item">
        {name} <PokeID>#{id}</PokeID>
      </CardItem>

      {/* Pokemon's type info */}
      {type[1] ? (
        <CardItem as="h3" className="card-item">
          {type[0]} / {type[1]}{" "}
        </CardItem>
      ) : (
        <h3 className="card-item">{type}</h3>
      )}

      <CardItem as="h4" className="card-item">
        Height: {height} <br />
        Weight: {weight}
      </CardItem>

      {/* image of each pokemon */}
      <PokeImageContainer as="figure" className="card-item">
        {/* Displays either the shiny or normal form of the pokemon based on user preference */}
        {isShiny ? (
          <PokeImage
            data-testid="shiny"
            src={`https://www.serebii.net/Shiny/SWSH/${num}.png`}
            alt={name}
          />
        ) : (
          <PokeImage
            data-testid="not-shiny"
            src={`https://www.serebii.net/swordshield/pokemon/${num}.png`}
            alt={name}
          />
        )}
      </PokeImageContainer>

      <CardItem as="blockquote" className="card-item">
        {dexEntry}
      </CardItem>
    </PokemonCard>
  );
};

export default SinglePokemon;
