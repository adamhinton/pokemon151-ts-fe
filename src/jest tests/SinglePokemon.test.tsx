import { render, screen } from "@testing-library/react";
import SinglePokemon from "../components/SinglePokemon";
import pokemonData, { PokemonData } from "../data";
import renderer from "react-test-renderer";

jest.mock("../data", (): PokemonData => {
  return [
    {
      id: 1,
      num: "001",
      name: "Testname123",
      type: ["TestType1", "TestType2"],
      height: "0.0001m",
      weight: "0.0001kg",
      dexEntry: "This is a test dex entry",
    },
  ];
});

test("[1] renders without errors", () => {
  render(<SinglePokemon poke={pokemonData[0]} isShiny={true} />);
});

test("[2] Displays correct Pokemon info on screen", () => {
  render(<SinglePokemon poke={pokemonData[0]} isShiny={true} />);

  const name = screen.getByText(/testname123/i);
  const type = screen.getByText(/TestType1/i);
  const dexEntry = screen.getByText(/This is a test dex entry/i);
  const height = screen.getByText(/.0001m/i);
  const weight = screen.getByText(/0.0001kg/i);

  expect(name).toBeVisible();
  expect(type).toBeVisible();
  expect(dexEntry).toBeVisible();
  expect(height).toBeVisible();
  expect(weight).toBeVisible();
});

test("[3] Matches snapshot from 9.16.22 --- this is before styling has been added", () => {
  const component = renderer.create(
    <SinglePokemon poke={pokemonData[0]} isShiny={true} />
  );

  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
