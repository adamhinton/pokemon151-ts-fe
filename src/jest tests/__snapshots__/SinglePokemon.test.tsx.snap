// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[3] Matches snapshot from 9.16.22 --- this is before styling has been added 1`] = `
<article
  className="sc-bczRLJ qcpOQ"
  data-testid="pokemon-card"
>
  <h2
    className="poke-name card-item"
  >
    Testname123
     
    <span
      className="dex-id"
    >
      #
      1
    </span>
  </h2>
  <h3
    className="poke-type card-item"
  >
    TestType1
     / 
    TestType2
     
  </h3>
  <h4
    className="height-weight card-item"
  >
    Height: 
    0.0001m
     
    <br />
    Weight: 
    0.0001kg
  </h4>
  <figure
    className="img-container card-item"
  >
    <img
      alt="Testname123"
      className="poke-img"
      data-testid="shiny"
      src="https://www.serebii.net/Shiny/SWSH/001.png"
    />
  </figure>
  <blockquote
    className="card-item"
  >
    This is a test dex entry
  </blockquote>
</article>
`;
